proto/topsql_agent.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package tipb;
+
+option java_multiple_files = true;
+option java_package = "com.pingcap.tidb.tipb";
+
+import "gogoproto/gogo.proto";
+
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.sizer_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+
+// TopSQLAgent is the persistent agent service for TopSQL records
+service TopSQLAgent {
+  // CollectCPUTime is called periodically (e.g. per minute) to save the in-memory TopSQL records
+  rpc CollectCPUTime(stream CollectCPUTimeRequest) returns (CollectCPUTimeResponse) {}
+}
+
+message CollectCPUTimeRequest {
+  string sql_digest = 1;
+  string plan_digest = 2;
+  repeated uint64 timestamp_list = 10;  // timestamp in second
+  repeated uint32 cpu_time_ms_list = 11;  // this can be greater than 1000 when counting concurrent running SQL queries
+  string normalized_sql = 20;  // SQL string with sensitive fields trimmed, potentially > 1M
+  string normalized_plan = 21;  // potentially > 1M
```

Comment:
Should add `bool is_internal = 22` field? @breeswish PTAL

Reply:
done